src/SpeakDirectiveParser.js

```diff
@@ -0,0 +1,56 @@
+const httpMessageParser = require("http-message-parser");
+const { hasIn } = require("immutable");
+const util = require("util");
+
+import IllegalArgumentError from "./errors/IllegalArgumentError";
+
+/**
+ * This method parses the multi-part AVS responses and extracts the
+ * string representing Alexa's response. It strips away other information
+ * like binary audio data, ssml tags etc.
+ *
+ * @param {String} alexaRawResponse The multi-part response from AVS. This
+ * should not be empty or undefined.
+ */
+export function extractAlexaTextResponse(alexaRawResponse) {
+  if (!alexaRawResponse) {
+    throw new IllegalArgumentError(
+      "The response to be parsed cannot be empty. Input: " + alexaRawResponse
+    );
+  }
+
+  const parsedResponse = httpMessageParser(alexaRawResponse);
+  if (
```

Comment:
Why dont you extract extractive avsDirective from parsed Response into a private method? That way you can test other failing cases (eg. parseArgument undefined or empty) which you failed to capture as of today.

Reply:
That feels like completely giving up and is a workaround anyways. At some point, I want to figure out how to mock httpParser for real. 

How about a TODO that says, we need to add tests by mocking httpMessageParser? It should be safe because httpMessageParser never returns empty response or empty response.multipart.body and it is a library that we can upgrade whenever we want to.

Reply:
Okay.

Reply:
Done.